src/datasources/balances-api/safe-balances-api.service.ts

```diff
@@ -108,10 +114,54 @@ export class SafeBalancesApi implements IBalancesApi {
     await this.cacheService.deleteByKey(key);
   }
 
-  /**
-   * No fiat prices are available from this provider.
-   */
-  getFiatCodes(): string[] {
-    return [];
+  async getFiatCodes(): Promise<string[]> {
+    return this.coingeckoApi.getFiatCodes();
+  }
+
+  private async _mapBalances(
+    balances: Balance[],
+    fiatCode: string,
+  ): Promise<Balance[]> {
+    const tokenAddresses = balances
+      .map((balance) => balance.tokenAddress)
+      .filter((address): address is string => address !== null);
+
+    const prices = await this.coingeckoApi.getTokenPrices({
+      chainId: this.chainId,
+      fiatCode,
+      tokenAddresses,
+    });
+
+    return await Promise.all(
+      balances.map(async (balance) => {
+        const tokenAddress = balance.tokenAddress?.toLowerCase() ?? null;
+        let price: number | null;
+        if (tokenAddress === null) {
+          price = await this.coingeckoApi.getNativeCoinPrice({
+            chainId: this.chainId,
+            fiatCode,
+          });
+        } else {
+          const found = prices.find((assetPrice) => assetPrice[tokenAddress]);
+          price = found?.[tokenAddress]?.[fiatCode.toLowerCase()] ?? null;
```

Comment:
It is guaranteed that `tokenAddress` is in lowercase, right? (it should have been be assured [in this line](https://github.com/safe-global/safe-client-gateway/blob/b6b8696b71591a4d6656f2f0a338190dbf260ede/src/datasources/balances-api/safe-balances-api.service.ts#L137)) 

Maybe I'm missing something 🤔

Reply:
I've added a lowercase conversion for all the addresses returned by `getTokenPrices` in [865d23f](https://github.com/safe-global/safe-client-gateway/pull/1175/commits/865d23f15de6eb55f5c7a09865307799931d80e9)

Hopefully, once we implement a general solution to checksum all addresses across the project, all this would be simplified also. 

Btw, I've also renamed back `CoingeckoAssetPrice` to `AssetPrice`, it was a mistake to prefix it with `Coingecko*`, because this entity is also used here in `SafeBalancesApi`, outside the Coingecko-specific context.